Automatic (derivatives ) for the people and a song of ice and fire 
Deadline:07_09_25
group number 12: 
 Amund Romseland Brevik
 Baraa Nidal Ghannoum
 Magnus Helvik Forsdahl 

 








Abstract: 



Introduction: 

exercise 1: 


1.1

In [1]:
import sys
sys.float_info

sys.float_info(max=1.7976931348623157e+308, max_exp=1024, max_10_exp=308, min=2.2250738585072014e-308, min_exp=-1021, min_10_exp=-307, dig=15, mant_dig=53, epsilon=2.220446049250313e-16, radix=2, rounds=1)

max=1.7976931348623157e+308 -
This is the largest floating point number that python can represent. Any attempt to go higher will result in inf.
A double uses 1 bit for sign, 11 bits for exponent, 52 bits for fraction (mantissa).

min=2.2250738585072014e-308 - This is the smallest positive normalized floating-point number. The minimum exponent for normalized numbers is −1022. 1.0 × 2^-1022 ≈ 2.2250738585072014 × 10^-308

epsilon = 2.220446049250313e-16 - The difference between 1.0 and the next representable floating-point value. Double precision has 53 bits of precision. So the spacing between floating-point numbers around 1.0 is 2^-52. So if you add anything smaller than epsilon to 1.0, rounding will round it equal to 1.0

exercise 2: 

exercise 3: 

exercise 4:

exercise 5: 

self reflections: 

References: 